src/components/ColorPicker/components/TextPicker/TextPicker.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {useCallback, useEffect, useState} from 'react';
+
+import {classNames} from '../../../../utilities/css';
+import {HSBColor} from '../../../../utilities/color-types';
+import {
+  normalizeColorString,
+  expandHex,
+  hsbToHex,
+  hsbToString,
+  rgbStringToHex,
+  nameToHex,
+  hexToHsb,
+} from '../../../../utilities/color-transformers';
+import {
+  isColorName,
+  isHexString,
+  isHashlessHex,
+  isRgbString,
+} from '../../../../utilities/color-validation';
+import {useI18n} from '../../../../utilities/i18n';
+import {TextField} from '../../../TextField';
+import styles from '../../ColorPicker.scss';
+
+export interface TextPickerProps {
+  color: HSBColor;
+  allowAlpha?: boolean;
+  onChange(hex: string): void;
+}
+
+export function TextPicker({color, allowAlpha, onChange}: TextPickerProps) {
+  const i18n = useI18n();
+
+  const [text, setText] = useState('');
+  const [lastValidValue, setLastValidValue] = useState('');
```

Comment:
We could get *probably*(this is an asumption) get away with this being a ref since we'll never need it to update a component (cause a re-render)

Reply:
Yep, you're right, I'll change it.